src/cfnlint/rules/resources/iam/ResourcePolicy.py

```diff
@@ -0,0 +1,136 @@
+"""
+  Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy of this
+  software and associated documentation files (the "Software"), to deal in the Software
+  without restriction, including without limitation the rights to use, copy, modify,
+  merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+  permit persons to whom the Software is furnished to do so.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+  INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+  PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+from cfnlint import CloudFormationLintRule
+from cfnlint import RuleMatch
+
+
+class ResourcePolicy(CloudFormationLintRule):
+    """Check if IAM Resource Policy JSON is correct"""
+    id = 'E3022'
+    shortdesc = 'Check IAM Resource Policies syntax'
+    description = 'See if the elements inside an IAM Resource policy ' + \
+                  'are configured correctly.'
+    source_url = 'https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html'
+    tags = ['properties', 'iam']
+
+    def __init__(self):
+        """Init"""
+        self.resources_and_keys = {
+            'AWS::SNS::TopicPolicy': 'PolicyDocument',
+            'AWS::S3::BucketPolicy': 'PolicyDocument',
+            'AWS::KMS::Key': 'KeyPolicy',
+            'AWS::SQS::QueuePolicy': 'PolicyDocument',
+            'AWS::ECR::Repository': 'RepositoryPolicyText',
+            'AWS::Elasticsearch::Domain': 'AccessPolicies',
+        }
+        for resource_type in self.resources_and_keys:
+            self.resource_property_types.append(resource_type)
+
+    def check_policy_document(self, value, path, cfn):
+        """Check policy document"""
+        matches = list()
+
+        valid_keys = [
+            'Version',
```

Comment:
We should enforce the most recent policy version ('2012-10-17') in a constant, per: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_version.html